src/main/kotlin/com/github/avrokotlin/avro4k/Avro.kt

```diff
@@ -254,6 +274,51 @@ class Avro(
       )
    }
 
+   /**
+    * Writes an instance of <T> using the given or derived writer [Schema].
+    * This method will use the [AvroEncodeFormat.SingleObject] format.
+    *
+    * @param serializer serialization strategy (as you get using `MyClass.serializer()`)
+    * @param writerSchema optional writer schema, if not given, it is derived from the serializer
+    * @param value the instance of type <T> we want to serialize.
+    *
+    * @return the written instance will be returned as a single object encoded [ByteArray] containing the schema fingerprint.
+    */
+   fun <T> encodeToSingleObject(
```

Comment:
I wouldn't add new decode and encode methods to the Avro class. Other encodings also do not have methods on this level. The only exception has been made for the "data" encoding. I'm thinking of deprecating the data methods to have a consistent API. So adding new methods would add new methods that need to be deprecated ^^.

Reply:
So for now, just using the in/out stream builders is the API? ok, can do.